src/dispatch/participant/flows.py

```diff
@@ -13,57 +15,75 @@
 
 log = logging.getLogger(__name__)
 
+Subject = TypeVar("Subject", Case, Incident)
+
 
 def add_participant(
```

Comment:
This feels like we may want to eventually, break up into `add_incident_participant` and `add_case_participant`. Since the overlap is high. I am okay leaving them combined for now.

Reply:
I'd be okay with breaking it out. The function can be hard to follow/read (for me) because of it handling two types.